cli/args/mod.rs

```diff
@@ -1227,6 +1227,14 @@ impl CliOptions {
   pub fn default_npm_caching_strategy(&self) -> NpmCachingStrategy {
     if self.flags.unstable_config.npm_lazy_caching {
       NpmCachingStrategy::Lazy
+    } else if matches!(
```

Comment:
drive by fix, it was using the eager caching strategy for these subcommands which causes `sync_resolution_with_fs` to run twice in some situations. Effectively we had regressed #25782 . This fixes it